ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlAuthorizationProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.security;
+
+import io.confluent.ksql.util.KsqlException;
+
+import java.security.Principal;
+
+/**
+ * Interface that provides authorization to KSQL.
+ */
+public interface KsqlAuthorizationProvider {
+  /**
+   * Checks if the user is authorized to access the endpoint.
+   *
+   * @param user The user who is requesting access to the endpoint
+   * @param method The endpoint method used, i.e. POST, GET, DELETE
+   * @param path The endpoint path to access, i.e. "/ksql", "/ksql/terminate", "/query"
+   * @throws KsqlException for access denied or any other authorization error
+   */
+  void checkEndpointAccess(Principal user, String method, String path) throws KsqlException;
```

Comment:
I think access denied errors merit a more specific exception type.

Reply:
I can use the Kafka `AuthorizationException` for it, what do you think? or should I create our own `KsqlAuthorizationException`?

Also, I want to let users know that other exceptions can be thrown for other errors, perhaps having both Authorization and KsqlException?